libs/test/SalesforceSDKTest/src/com/salesforce/androidsdk/ClearupOrgData.java

```diff
@@ -0,0 +1,124 @@
+
+import com.salesforce.androidsdk.auth.HttpAccess;
+import com.salesforce.androidsdk.auth.OAuth2;
+import com.salesforce.androidsdk.auth.OAuth2.TokenEndpointResponse;
+import com.salesforce.androidsdk.rest.RestClient;
+import com.salesforce.androidsdk.rest.RestClient.ClientInfo;
+import com.salesforce.androidsdk.rest.RestRequest;
+import com.salesforce.androidsdk.rest.RestRequest.RestMethod;
+import com.salesforce.androidsdk.rest.RestResponse;
+
+import java.net.HttpURLConnection;
+import java.net.URI;
+/**
+ * Do org data cleanup
+ *
+ */
+public class ClearupOrgData extends InstrumentationTestCase {
+
+    private ClientInfo clientInfo;
+    private HttpAccess httpAccess;
+    private RestClient restClient;
+    private String authToken;
+    public static final String TEST_FIRST_NAME = "firstName";
+    public static final String TEST_LAST_NAME = "lastName";
+    public static final String TEST_DISPLAY_NAME = "displayName";
+    public static final String TEST_EMAIL = "test@email.com";
+    public static final String TEST_PHOTO_URL = "http://some.photo.url";
+    public static final String TEST_THUMBNAIL_URL = "http://some.thumbnail.url";
+    private final int MaxTransactCount = 100;
+    private final String EXPECTED_DELETE = "ExpectedDeletes";
+
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        TestCredentials.init(getInstrumentation().getContext());
+        httpAccess = new HttpAccess(null, "dummy-agent");
+        TokenEndpointResponse refreshResponse = OAuth2.refreshAuthToken(httpAccess, new URI(TestCredentials.INSTANCE_URL), TestCredentials.CLIENT_ID, TestCredentials.REFRESH_TOKEN);
+        authToken = refreshResponse.authToken;
+        clientInfo = new ClientInfo(TestCredentials.CLIENT_ID,
+        		new URI(TestCredentials.INSTANCE_URL),
+        		new URI(TestCredentials.LOGIN_URL),
+        		new URI(TestCredentials.IDENTITY_URL),
+        		TestCredentials.ACCOUNT_NAME, TestCredentials.USERNAME,
+        		TestCredentials.USER_ID, TestCredentials.ORG_ID, null, null,
+                TEST_FIRST_NAME, TEST_LAST_NAME, TEST_DISPLAY_NAME, TEST_EMAIL, TEST_PHOTO_URL, TEST_THUMBNAIL_URL);
+        restClient = new RestClient(clientInfo, authToken, httpAccess, null);
+    }
+
+    /**
+     * Testing a customer apex api call.
+     * @throws Exception
+     */
+    public void testcleanOrgData() throws Exception {
+        int stop = 0;
+        int expectedCount = 0;
+        String requestFormat = "/services/apexrest/Cleanup?start=%d&stop=%d";
+        do {
+            int start = stop;
+            stop = stop + MaxTransactCount;
+
+            RestResponse response = restClient.sendSync(new RestRequest(RestMethod.DELETE, String.format(requestFormat, start, stop), null));
+            checkResponse(response, HttpURLConnection.HTTP_OK, false);
+            expectedCount = response.asJSONObject().getInt(EXPECTED_DELETE);
+        } while (expectedCount >= MaxTransactCount);
```

Comment:
The apex class needs to know the number of records needs to be deleted next time from the last api call, via batch call, how to figure out how many records need to be deleted? Could you show me more details or Is there any example that I can take a look?